xc/device.ml

```diff
@@ -2319,7 +2319,7 @@ module Backend = struct
     module Monitor = struct
       module Epoll = Core.Linux_ext.Epoll
       module Flags = Core.Linux_ext.Epoll.Flags
-      let create () = (Core.Or_error.ok_exn Epoll.create) ~num_file_descrs:1001 ~max_ready_events:1
+      let create () = (Core.Or_error.ok_exn Epoll.create) ~num_file_descrs:4096 ~max_ready_events:1
```

Comment:
Instead of hardcoding the number this can also be found out at runtime, if you put this into `Monitor` and use it in `create`:
`let num_file_descrs = Core.Unix.RLimit.((get num_file_descriptors).cur)`

The context here is that xenopsd has a raised ulimit of 4096, and file descriptors can get assigned higher than 1001, but should that limit change again in the future it would be good if this code automatically adapted.

Reply:
@minli1 when using Core.Unix.RLimit above, during dev-tests please make sure that you debug-print num_file_descrs to ascertain it's 4096.

Reply:
That is a good point, we should always print that number into the logs at debug level, to make debugging easier in the future.

Reply:
Thanks all. @edwintorok @mg12 
I updated the code now. Build and xenRT tests in progress, will add the logs here with debug print of the FD limit.

If the `num_file_descriptors` return `Infinity`, then I set the num_file_descrs to Int.max_value. Is there any concern?